spec/src/main/resources/META-INF/smithy/bsp/extensions/rust.smithy

```diff
@@ -0,0 +1,393 @@
+}
+
+list RustPackages {
+    member: RustPackage
+}
+
+/// A `crate` is the smallest amount of code that the Rust compiler considers at a time.
+/// It can come in one of two forms: a binary crate or a library crate.
+/// `Binary crates` are programs you can compile to an executable that you can run,
+/// such as a command-line program or a server.
+/// Each must have a function called main that defines what happens when the executable runs.
+/// `Library crates` don’t have a main function, and they don’t compile to an executable.
+/// Instead, they define functionality intended to be shared with multiple projects.
+///
+/// A `package` is a bundle of one or more crates that provides a set of functionality.
+/// It contains a Cargo.toml file that describes how to build those crates.
+/// A package can contain many binary crates, but at most only one library crate.
+/// However, it must contain at least one crate, whether that’s a library or binary crate.
+structure RustPackage {
+    /// The package’s unique identifier
+    @required
+    id: String
+    /// The package's root path.
+    @required
+    rootUrl: URI
+    /// The name of the package.
+    @required
+    name: String
+    /// The version of the package.
+    @required
+    version: String
+    /// Defines a reason a package is in a project.
+    @required
+    origin: RustPackageOrigin
+    /// Code edition of the package.
+    @required
+    edition: RustEdition
+    /// The source ID of the dependency, `null` for the root package and path dependencies.
+    source: String
+    /// Correspond to source files which can be compiled into a crate from this package.
+    /// Contains only resolved targets without conflicts.
+    @required
+    targets: RustTargets
+    /// Same as `targets`, but contains all targets from this package.
+    /// `targets` should be the subset of `allTargets`.
+    @required
+    allTargets: RustTargets
+    /// Set of features defined for the package.
+    /// Each feature maps to an array of features or dependencies it enables.
+    /// The entry named "default" defines which features are enabled by default.
+    @required
+    features: RustFeatures
+    /// Array of features enabled on this package.
+    @required
+    enabledFeatures: RustPackageEnabledFeatures
+    /// Conditional compilation flags that can be set based on certain conditions.
+    /// They can be used to enable or disable certain sections of code during the build process.
+    cfgOptions: RustCfgOptions
+    /// Environment variables for the package.
+    env: RustEnvironmentVariables
+    /// An absolute path which is used as a value of `OUT_DIR` environmental
+    /// variable when compiling current package.
+    outDirUrl: URI
+    /// File path to compiled output of a procedural macro crate.
+    /// Procedural macros are macros that generate code at compile time.
+    /// Contains files with file extensions: `.dll`, `.so` or `.dylib`.
+    procMacroArtifact: String
```

Comment:
why not a URI?

Reply:
intellij-rust doesn't want URI here

Reply:
Can't we map/extract it on the client side to keep the protocol consistent?

Reply:
We can do that, I fixed it to URI